pulsar-jms-tracing/src/main/java/com/datastax/oss/pulsar/jms/tracing/BrokerTracing.java

```diff
@@ -0,0 +1,553 @@
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.jms.tracing;
+
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.TraceLevel;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getCommandDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getConnectionDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getConsumerDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getEntryDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getMessageMetadataDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getProducerDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getPublishContextDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.getSubscriptionDetails;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.trace;
+import static com.datastax.oss.pulsar.jms.tracing.TracingUtils.traceByteBuf;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.netty.buffer.ByteBuf;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.bookkeeper.mledger.Entry;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.intercept.BrokerInterceptor;
+import org.apache.pulsar.broker.service.Consumer;
+import org.apache.pulsar.broker.service.Producer;
+import org.apache.pulsar.broker.service.ServerCnx;
+import org.apache.pulsar.broker.service.Subscription;
+import org.apache.pulsar.broker.service.Topic;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.common.api.proto.BaseCommand;
+import org.apache.pulsar.common.api.proto.CommandAck;
+import org.apache.pulsar.common.api.proto.MessageIdData;
+import org.apache.pulsar.common.api.proto.MessageMetadata;
+import org.apache.pulsar.common.intercept.InterceptException;
+import org.apache.pulsar.common.naming.TopicName;
+import org.jetbrains.annotations.NotNull;
+
+@Slf4j
+public class BrokerTracing implements BrokerInterceptor {
+
+  public enum EventReasons {
+    ADMINISTRATIVE,
+    COMMANDS,
+    MESSAGE,
+    TRANSACTION,
+    SERVLET,
+  }
+
+  private static final TraceLevel defaultTraceLevel = TraceLevel.BASIC;
+
+  private final Set<EventReasons> jmsTracingEventList = new HashSet<>();
+  private TraceLevel traceLevel = defaultTraceLevel;
+  private int maxBinaryDataLength = 256;
+  private int cacheTraceLevelsDurationSec = 10;
```

Comment:
I think that is better to set this to a higher value, like 30 minutes, in order to reduce the impact on executing blocking operations in the hotpaths